packages/sdk/test/integration/quantum-policy.test.ts

```diff
@@ -0,0 +1,170 @@
+
+describe('Quantum encryption policies', () => {
+
+    let publisherIdentity: Identity
+    let subscriberIdentity: Identity
+    let quantumPublisher: StreamrClient
+    let quantumSubscriber: StreamrClient
+    let nonQuantumClient: StreamrClient
+    let stream: Stream
+    let environment: FakeEnvironment
+
+    beforeEach(async () => {
+        environment = new FakeEnvironment()
+
+        publisherIdentity = MLDSAKeyPairIdentity.generate()
+        subscriberIdentity = MLDSAKeyPairIdentity.generate()
+
+        // Client for setting up the stream
+        nonQuantumClient = environment.createClient({
+            auth: {
+                identity: EthereumKeyPairIdentity.generate(),
+            }
+        })
+        // nonQuantumClient also has pub/sub permissions to stream because it created the stream
+        stream = await createTestStream(nonQuantumClient, module)
+        await stream.grantPermissions({ permissions: [StreamPermission.PUBLISH], userId: await publisherIdentity.getUserIdString() })
+        await stream.grantPermissions({ permissions: [StreamPermission.SUBSCRIBE], userId: await subscriberIdentity.getUserIdString() })
+    })
+
+    afterEach(async () => {
+        await environment.destroy()
+    })
+
+    describe('pubsub under strict quantum settings', () => {
+
+        let sub: Subscription
+
+        beforeEach(async () => {
+            // Clients for pub/sub (can't make transactions)
+            quantumPublisher = environment.createClient({
+                auth: {
+                    identity: publisherIdentity,
+                },
+                encryption: {
+                    requireQuantumResistantSignatures: true,
+                    requireQuantumResistantKeyExchange: true,
+                    requireQuantumResistantEncryption: true,
+                }
+            })
+            quantumSubscriber = environment.createClient({
+                auth: {
+                    identity: subscriberIdentity,
+                },
+                encryption: {
+                    requireQuantumResistantSignatures: true,
+                    requireQuantumResistantKeyExchange: true,
+                    requireQuantumResistantEncryption: true,
+                }
+            })
+
+            sub = await quantumSubscriber.subscribe({
+                streamId: stream.id,
+            })
+        })
+
+        it('works between quantum identities', async () => {
+            const sub = await quantumSubscriber.subscribe({
+                streamId: stream.id,
+            })
+            const testMsg = Msg()
+            await quantumPublisher.publish(stream.id, testMsg)
+            const received = []
+            for await (const msg of sub) {
+                received.push(msg.content)
+                if (received.length === 1) {
+                    break
+                }
+            }
+            expect(received).toEqual([testMsg])
```

Comment:
It is possible to collect the messages easily with the `collect` helper function:

```suggestion
            const received = await collect(sub, 1)
            expect(received.map((m) => m.content)).toEqual([testMsg])
```

Reply:
Very handy! 👍